### PR TITLE
Add description about the plugin resources

### LINKED ISSUE #3: More description of plugin externals
Consider to add the following:
* a description of methods used by the wallet and provided by assets. The description should state which of them should be overridden if an application has another way to send tokens, get events about the token transfer etc.
* a description of methods used by the exchange plugin and provided by mediators. The description is to understand which functionality must be overridden if mediators contracts implemented by the application differ from our mediators.

Probably something more should be added. Just think how a new developer which has no much information about BW internals and internals of our plugin could rapidly develop another version of plugin based on the documentation provide in the sample repo.

### PR DESCRIPTION
**Pull request by patitonar**
Closes #3

Once it is merged, the docs at https://docs.tokenbridge.net/amb-bridge/create-a-burner-wallet-plugin-for-your-amb-bridge-extension will be updated with this content too

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,107 @@
...
 
-#### Project structure
-There are two main folders in the project:
+`Pair` is an abstract class that defines the interface for trading pairs. If you would like to add custom trading functionality, you can extend this class and implement it's methods:
+- `estimateAtoB`: Indicates how to estimate the B Asset the user will receive for the amount of A Asset. For example Fees could be considered here.
+- `estimateBtoA`: Indicates how to estimate the A Asset the user will receive for the amount of B Asset. For example Fees could be considered here.
+- `exchangeAtoB`: Indicates how to send A Asset to be exchanged for B Asset. Here you can add logic to detect exchange finalization for example for a more complex exchange operation like bridges.
+- `exchangeBtoA`: Indicates how to send B Asset to be exchanged for A Asset. Here you can add logic to detect exchange finalization for example for a more complex exchange operation like bridges.
+
+### Project structure
```
Comment (akolotov) [inline README.md]: Consider to move details about the TokenBridge plugin with the link to the repo before the project structure section: which components (`ERC677Asset`, `NativeMediatorAsset`, `Mediator`) it provides to be used by adopters, which interfaces these components provides.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,107 @@
...
 
-### Clone the sample repo 
-Clone the sample repo to create a plugin project that uses the TokenBridge plugin. This sample repo containes a ready to publish plugin for a ERC677 to ERC677 bridge extension, but you can later modify it to define your own resources.
+The `ERC20Asset` class extends from the `Asset` class. The `Asset` class provides several methods to allow the wallet interact with the token.
+The following are the most important to pay attention and that could be overridden if your token behaves differently.
+- `getBalance`: Indicates how to get the balance of your asset.
+- `startWatchingAddress`: Indicates how to listen to value transfers incoming into your account.
+- `_send`: Indicates how to send value to another account.
+
+#### Exchange Plugin
```
Comment (akolotov) [inline README.md:42]: Extend the section with links to examples of the exchange plugins: xDai, Uniswap etc. Probably, it makes sense to add a short note for every example what is most interesting to pay attention (two-step interaction, transfer completion etc.)

File: README.md [hunk 3]
```diff
@@ -1,24 +1,107 @@
...
+
+The class `ERC677Asset` extends from `ERC20Asset` and overrides some methods:
+- `_send`: In this case the method `transferAndCall` is used instead of the `transfer` method.
+- `startWatchingAddress`: Add logic to correctly track the transfer events related to the wallet account.
+- `getTx`: Overrides the logic to return the stored information about a transfer transaction to avoid errors on the information displayed.
+
+The class `Mediator` extend from some other classes that has `Pair` as base. It implements the previous mentioned methods to estimate and exchange the assets. It also implements some methods to detect the exchange finalization.
+Here are some details of the implemented methods:
+- `estimateAtoB`: Get the fee and calculates the amount of B Asset the user will receive.
+- `estimateBtoA`: Get the fee and calculates the amount of A Asset the user will receive.
```
Comment (akolotov) [inline README.md]: We need to specify somewhere that there is a special field that can be used to notify an user about exchange properties (we use this field to notify about the transaction fees).

File: README.md [hunk 4]
```diff
@@ -1,24 +1,107 @@
...
+
+```
+import { Stake, xStake, StakeBridge } from 'my-plugin'
+```
+
+To be able to use and to display the balances for Stake and xStake it is necessary to list them in the list of assets as follow:
+```
+const core = new BurnerCore({
+    ...
+  assets: [Stake, xStake]
```
Comment (akolotov) [inline README.md:123]: Please also specify some where about gateways that are used to automatically choose a Web3 provider. Describe shortly a way how the provider is being chosen.

**Review by akolotov — CHANGES_REQUESTED** [review]
Provided suggestions for the information re-arrangement.

**Comment by patitonar** [comment]
Thanks for the suggestions! I updated the documentation

**Review by akolotov — APPROVED** [review]
The documentation can be published now on tokenbridge.net as well.

**Pull request merged**